src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs

```diff
@@ -779,8 +794,14 @@ protected override void Dispose(bool disposing)
                     // In the async case the underlying stream is the async buffered stream, so we have to flush that explicitly.
                     if (this.asynchronousOutputStream != null)
                     {
+#if NETSTANDARD1_1
                         this.asynchronousOutputStream.FlushSync();
                         this.asynchronousOutputStream.Dispose();
+#else
+                        this.asynchronousOutputStream.Flush();
+                        // We are working with a BufferedStream here. We flushed it already, so there is nothing else to dispose. And it would dispose the 

```

Comment:
Does flushing a `BufferedStream` also disposed the stream? This isn't mentioned in the docs and doesn't appear to be the case in the [source code](https://source.dot.net/#System.Private.CoreLib/BufferedStream.cs,261). I think `Dispose` the `BufferedStream` will automatically flush it, but not the other way around. I think it would be safer to have an explicit `Dispose`/`DisposeAsync` .

Reply:
@habbes It does not. However, the .NET BufferStream always disposes of the the underlying stream when `Dispose` is called and that leads to the situation where the inner stream is closed twice. And that breaks some unit tests. 

Since both the AsyncBufferedStream and the .NET BufferedStream don't do anything interesting in their `Dispose` methods anyway besides flushing the the stream, I decided not to dispose if it here.

Reply:
Hi @henning-krause  just want to verify I understand what you're implying. The reason you don't want to dispose the buffered stream is because it also disposes the inner stream. In this case the inner stream is `this.messageOutputStream`. The `messageOutputStream` is being disposed a few lines below so we don't need to dispose the BufferedStream because it would lead to the `messageOutputStream` being disposed twice, and besides disposing the inner stream, it does nothing interesting since it has already been flushed.

Is this a correct understanding of  your statements?

If that is the case, then I agree you do have a point. But the fact that relies on the knowlede of the internals of  `BufferedStream.Dispose()`, that it does does nothing interesting, bothers me a bit. I think the fact tests fail when `messageOutputStream.Dispose()` is called twice also bothers me a bit since `Dispose()` is supposed to be idempotent. That said, that is not caused by this PR and I don't think it should block it.